plugins/UM3NetworkPrinting/src/SendMaterialJob.py

```diff
@@ -1,99 +1,191 @@
 # Copyright (c) 2018 Ultimaker B.V.
 # Cura is released under the terms of the LGPLv3 or higher.
+import json
+import os
+import urllib.parse
+from typing import Dict, TYPE_CHECKING, Set
 
-import json #To understand the list of materials from the printer reply.
-import os #To walk over material files.
-import os.path #To filter on material files.
-from PyQt5.QtNetwork import QNetworkReply, QNetworkRequest #To listen to the reply from the printer.
-from typing import Any, Dict, Set, TYPE_CHECKING
-import urllib.parse #For getting material IDs from their file names.
+from PyQt5.QtNetwork import QNetworkReply, QNetworkRequest
 
-from UM.Job import Job #The interface we're implementing.
+from UM.Job import Job
 from UM.Logger import Logger
-from UM.MimeTypeDatabase import MimeTypeDatabase #To strip the extensions of the material profile files.
+from UM.MimeTypeDatabase import MimeTypeDatabase
 from UM.Resources import Resources
-from UM.Settings.ContainerRegistry import ContainerRegistry #To find the GUIDs of materials.
-
-from cura.CuraApplication import CuraApplication #For the resource types.
+from cura.CuraApplication import CuraApplication
+# Absolute imports don't work in plugins
+from .Models import ClusterMaterial, LocalMaterial
 
 if TYPE_CHECKING:
     from .ClusterUM3OutputDevice import ClusterUM3OutputDevice
 
+
 ##  Asynchronous job to send material profiles to the printer.
 #
 #   This way it won't freeze up the interface while sending those materials.
 class SendMaterialJob(Job):
+
     def __init__(self, device: "ClusterUM3OutputDevice") -> None:
         super().__init__()
-        self.device = device #type: ClusterUM3OutputDevice
+        self.device = device  # type: ClusterUM3OutputDevice
+        self._application = CuraApplication.getInstance()  # type: CuraApplication
```

Comment:
We only need to use `Application`, `CuraApplication` is unnecessary because you're not using one of its specialised functions.